### DISCUSSION TITLE
Any plan to add event like OnChange to TextField

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#5207] [Help] [answered]
Hi guys, I have a flow to invoke an api when the value in textfield been changed, and I'm not sure any plan or any workaround to do that in budibase?

**Comment by melohagan** [comment]
Hi @ryzam 

There is no way to do an onChange event on a text field presently. 

In your case, is it necessary to be calling an api every time a textfield changes? Seems like it could be quite intensive.

**Comment by xnohat** [comment]
Text field change can trigger a query update and update a table when text change in text field with an "ENTER"

**Reply by ryzam** [reply]
@xnohat any example how to listen to "ENTER" event?

**Reply by xnohat** [reply]
Create 
Text Field
Table Block

in setting of Table Block set a Query as data source , click wheel icon beside Data setting, fill query binding using data from Text Field

Now every change in Text Field will trigger a change in Query -> change in Table Block

![image](https://user-images.githubusercontent.com/4613808/161376844-53942eaa-b6cb-4756-84af-d9d9a35c5b42.png)

**Comment by samg666** [comment]
Same request, needs onClick/onChange event for text field and option picker.

**Comment by melohagan** [comment, marked as answer]
@ryzam @ggs331 

Hey guys, just letting you know this is going to be worked on: https://github.com/Budibase/budibase/issues/5392

Hopefully available for the next release!

**Reply by melohagan** [reply]
This feature has been released :) 
https://github.com/Budibase/budibase/discussions/5849

**Reply by samg666** [reply]
Thank you @melohagam , great features and I am running on it for few days.
